### PR TITLE
Add eeg reaper

### LINKED ISSUE #59: Add EEG reaper
Add EEG reaper

- Identify how group/project/subject/session-uid/acquisition-uid should be _"encoded"_
    - Current plan:
        - group/project/subject ~ directory structure
        - session/acquisition ~ filename prefix
- Implement new reaper
- Add tests

### PR DESCRIPTION
**Pull request by ambrussimon**
Closes #59 

Added EEG reaper with the feature set discussed so far.
Manually ran reaper in various scenarios and added unit tests as well.
Will rebase onto master once dicom_folder_sniper gets merged.

### CODE REVIEW

**Comment by larsoner** [comment]
As I mentioned in #59, there is already a reaper that does both MEG and EEG (any formats supported by MNE-Python). Why not use that?

**Comment by ambrussimon** [comment]
@Eric89GXL Hey, Thanks for mentioning MNE-Python! I looked around and found BrainVision file handling, which is really neat. The purpose of this reaper however is to simply poll the filesystem and upload EEGs it finds to scitran/core, for which there's currently no need to parse the data. If there's a similar feature to that in mne-python that could have been reused, could you please point me to it? Thanks!

**Comment by larsoner** [comment]
> I looked around and found BrainVision file handling, which is really neat.

Currently [7 EEG-specific formats are supported](https://mne-tools.github.io/dev/manual/io.html#importing-eeg-data) (with more EEG formats implicitly supported as they are part of MEG+EEG systems), so it's more general than that.

> The purpose of this reaper however is to simply poll the filesystem and upload EEGs it finds to scitran/core, for which there's currently no need to parse the data.

Don't you need to read in metadata, such as subject information, acquisition date, etc.? I can't remember the specs.

MNE itself doesn't do any scanning/polling, but should be useful for any such metadata queries. However, it seems like whatever EEG stuff is done here could/should probably be implemented to use the existing MEG (+EEG) reaper. I thought I wrote one anyway... maybe you just need to add file extensions?

**Comment by gsfr** [comment]
Thanks, @ambrussimon-invenshure. This all looks good to me. I like your `argparse` solution. I will try to finish up the dicom folder sniper PR today, so you'll need to rebase one more time.

**Comment by ambrussimon** [comment]
@gsfr The latest de-identify commits don't break anything on my end. Will rebase to master as soon as dicom_folder_sniper gets merged, then will merge right after the updated PR gets approved.

File: reaper/util.py
Comment (gsfr) [inline reaper/util.py]: Looks like a potential [tarbomb](https://en.wikipedia.org/wiki/Tar_(computing)#Tarbomb). Why not always keep the directory?

File: reaper/util.py
Comment (gsfr) [inline reaper/util.py]: If we end up needing this additional parameter, we should add it to the end of the list. As is, this breaks some of the other tools that rely on parameter positions rather than names.

File: reaper/eeg_reaper.py
Comment (gsfr) [inline reaper/eeg_reaper.py]: I think the `list()` call is superfluous here. And, generally, `filter()`, `map()`, and `reduce()` should be avoided in favor of more readable / pythonic list comprehensions. I'm surprised pylint isn't complaining about this.

`[fni for fni in filename_info if fni]`

File: reaper/eeg_reaper.py
Comment (gsfr) [inline reaper/eeg_reaper.py]: `.strip('.')` might be easier to read than `.replace('.', '')`.

File: reaper/eeg_reaper.py
Comment (gsfr) [inline reaper/eeg_reaper.py]: Per our "new convention" :wink:, all log statements now start with a capital letter.

Please apply throughout; I'm only noting this here.

**Review by gsfr — CHANGES_REQUESTED** [review]
@ambrussimon-invenshure #64 has landed. I took the liberty to rebase this branch onto master in order to reduce the diff.

Nice work. I left a couple of minor comments. After addressing those, I think we are ready for a test run. We may want to hold off on merging until after that.

File: reaper/util.py
Comment (ambrussimon) [reply reaper/util.py]: I found that the EEG viewer in the frontend _requires_ a zip where the .eeg and .vhdr are present at the root level (and that the zip filename ends with .eeg.zip). I assumed it was the decided format.

File: reaper/util.py
Comment (ambrussimon) [reply reaper/util.py]: Ah, thanks, fixed. I remember grepping for it, but forgot to follow up on the topic, as I also think there's a [broken usage](https://github.com/scitran/reaper/blob/master/reaper/gephysio.py#L62) in gephysio. (`metadata` and `outdir` is swapped)

**Comment by ambrussimon** [comment]
@gsfr Thanks for rebasing and commenting! Fixed everything, the only potentially open question is whether the zip structure expected by the viewer will change or not. For the time being, leaving the root-dir feature as-is.

File: reaper/util.py
Comment (gsfr) [reply reaper/util.py]: Thanks for catching that. We'll update the viewer, which was recently merged, btw, to match our standard convention of one directory level (https://github.com/flywheel-io/frontend/issues/1093).

Let's rip this out after the demo.

File: reaper/util.py
Comment (gsfr) [reply reaper/util.py]: Good catch on this, as well. gephysio has been unmaintained for a while. There is more than one problem with that line. I'll add a general issue and point specifically to this line.

**Pull request closed without merging**
